core/templates/services/contextual/url.service.spec.ts

```diff
@@ -143,18 +143,14 @@ describe('Url Service', () => {
   });
 
   it('should correctly retrieve classroom name from url', () => {
-    mockLocation.pathname = '/classroom/abcdefgijklm';
+    mockLocation.pathname = '/abcdefgijklm';
     expect(
       urlService.getClassroomNameFromUrl()
     ).toBe('abcdefgijklm');
-    mockLocation.pathname = '/classroom/class%20name';
+    mockLocation.pathname = '/class%20name';
     expect(
       urlService.getClassroomNameFromUrl()
     ).toBe('class name');
-    mockLocation.pathname = '/invalid/abcdefgijklm';
```

Comment:
There should be some replacement test for the invalid case, right?

Also, getClassroomNameFromUrl() should have some way to decide (perhaps based on page content or something that only exists within classroom pages) whether a page is really a classroom page.

Reply:
I've made changes.
RE 2nd para, since the only purpose of this function is to parse and extract the classroom name from the URL, shouldn't we only be checking if the structure is valid? 
Does it make sense to check if the classroom name is valid in this function?
The reason I'm asking is that the other functions in this URL services file, like getTopicNameFromLearnerUrl, or getTopicIdFromUrl all just check for structural validity.

Reply:
It's a good question and I'm not sure. @aks681 -- you have the most experience here, what do you think?

Reply:
@iamprayush Yeah, you can't test for structural validity here. Though, you could do something like, set a custom page context (for which, we already have functions in ContextService), in classroom page controller and maybe verify that in the function here? I can't think of anything else.
There may be  a delay there, between loading the page and setting the context, so not sure if you can throw an error, you'd need to try it.

Reply:
@aks681 Just to confirm that I've understood correctly, what you're suggesting is that I should set a custom entity context in the classroom page controller and then check if the entityID is one of the classroom names? Like so:
```typescript
ContextService.setCustomEntityContext(ENTITY_TYPE.CLASSROOM,
    ctrl.classroomName);
```
Shouldn't we instead directly check if the classroom name is present in the constant containing all classroom names?

Reply:
Yeah, but the constant containing classroom names is available only in the backend, right (in config_domain)? 
You could return that as well when fetching a classroom page, but that would be equivalent to using custom entities, since the classroom fetch function would only return anything if the classroom name exists.